flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/rules/AbstractCodeGeneratorCastRule.java

```diff
@@ -148,9 +147,10 @@ protected AbstractCodeGeneratorCastRule(CastRulePredicate predicate) {
                                     classCode)
                             .getConstructors()[0]
                             .newInstance(constructorArgs);
-        } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+        } catch (Throwable e) {
```

Comment:
IIRC I converted to `Throwable` to catch also `Error`s and not just `Exception`. I can revert if you want, but I think it's better to keep it this way because invoking this constructor should never fail, for any reason, and we need to catch it immediately if it fails in order to wrap it with the code associated.